From 7a13c962a909fc35cb04823caf4d508a292d0e65 Mon Sep 17 00:00:00 2001
From: Naiwen Yu <quic_naiwyu@quicinc.com>
Date: Mon, 8 May 2023 23:37:12 +0800
Subject: [PATCH] systemd: ignore VT swtich signal between host and container

Modification: Add judgement to avoid seat deactivated because of VT switch
singal cross between host and container under LXC.

Upstream-Status:Pending

Change-Id:85d554cc87c1042ef36b92a3c23bf29c9bd34b81
Signed-off-by: Naiwen Yu <quic_naiwyu@quicinc.com>
---
 src/login/logind-seat.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/login/logind-seat.c b/src/login/logind-seat.c
index 58912b85b3..ddd2a8fa95 100644
--- a/src/login/logind-seat.c
+++ b/src/login/logind-seat.c
@@ -378,6 +378,17 @@ int seat_active_vt_changed(Seat *s, unsigned vtnr) {
                         }
         }
 
+        /* VT change signal will cross in host and container under LXC,
+         * for example, host seat0 is active on VT2, then container VT7
+         * activated, host will get an inappropriate VT change signal,
+         * since seat0 not bind to VT7 in host, it would be set inactive.
+         * Need to add guard here to avoid seat inactivate after VT change.
+         * Notice: It only works for single seat use case!!!
+         */
+        if((new_active == NULL) && (s->active != NULL)) {
+            new_active = s->active;
+        }
+
         r = seat_set_active(s, new_active);
         manager_spawn_autovt(s->manager, vtnr);
 
-- 
2.25.1

